src/substrate/storageFetcher.ts

```diff
@@ -359,36 +362,37 @@ export class StorageFetcher extends IStorageFetcher<ApiPromise> {
         fee: b.bounty.fee.toString(),
         curatorDeposit: b.bounty.curatorDeposit.toString(),
         bond: b.bounty.bond.toString(),
+        description: description?.isSome
+          ? hexToString(description.unwrap().toString())
+          : undefined,
       } as ITreasuryBountyProposed);
 
       if (
-        b.bounty.status.isProposed ||
-        b.bounty.status.isNone ||
-        b.bounty.status.isCuratorProposed
-      )
-        return; // Return here, not progressed
-
-      events.push({
-        kind: EventKind.TreasuryBountyBecameActive,
-        bountyIndex: +b.index,
-      } as ITreasuryBountyBecameActive);
-      if (b.bounty.status.isActive) return;
-      if (b.bounty.status.isPendingPayout) {
+        !b.bounty.status.isProposed &&
+        !b.bounty.status.isNone &&
+        !b.bounty.status.isCuratorProposed
+      ) {
         events.push({
-          kind: EventKind.TreasuryBountyAwarded,
+          kind: EventKind.TreasuryBountyBecameActive,
           bountyIndex: +b.index,
-          value: b.bounty.value.toString(),
-          beneficiary: b.bounty.status.asPendingPayout.beneficiary.toString(),
-        } as ITreasuryBountyAwarded);
-        events.push({
-          kind: EventKind.TreasuryBountyClaimed,
-          bountyIndex: +b.index,
-          payout: b.bounty.value.toString(),
-          beneficiary: b.bounty.status.asPendingPayout.beneficiary.toString(),
-        } as ITreasuryBountyClaimed);
+        } as ITreasuryBountyBecameActive);
+        if (!b.bounty.status.isActive && b.bounty.status.isPendingPayout) {
+          events.push({
+            kind: EventKind.TreasuryBountyAwarded,
+            bountyIndex: +b.index,
+            value: b.bounty.value.toString(),
+            beneficiary: b.bounty.status.asPendingPayout.beneficiary.toString(),
+          } as ITreasuryBountyAwarded);
+          events.push({
+            kind: EventKind.TreasuryBountyClaimed,
+            bountyIndex: +b.index,
+            payout: b.bounty.value.toString(),
+            beneficiary: b.bounty.status.asPendingPayout.beneficiary.toString(),
+          } as ITreasuryBountyClaimed);
```

Comment:
the isPendingPayout state occurs before the bounty is claimed - does that make this incorrect?

Reply:
I didn't change the logic at all, just moved the conditionals around. Perhaps it's always been incorrect and we should look into it.

Reply:
Okay, fixed!